### ISSUE TITLE
don't report `Any` on certain function call parameters that aren't `Any`

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#318]
```py
def foo(l: list[Any]): ...

foo([1, 2, 3])  # error: Any expression
```

It it really an `Any` expression?

What do you think (comment below)

**Comment by DetachHead** [comment]
how is it not an `Any` expression?

**Comment by KotlinIsland** [comment]
hmm, this is a tricky one, but maybe you're right.

```py
from typing import Any

a: Any
a = [1, 2, 3] # error: has type "List[Any]"
```

**Comment by DetachHead** [comment]
I think both should be errors

Is the first one not an error because of those mypy issues you found where the any errors don't appear?

**Comment by KotlinIsland** [comment]
I think it's supposed to be intentional tbh

**Comment by DetachHead** [comment]
why?
